web-app/HeimScripts/heatmap/run.R

```diff
@@ -1,187 +1,254 @@
 library(reshape2)
 
-
-main <- function(max_rows=100, sorting="nodes"){
+main <- function(max_rows = 100, sorting = "nodes") {
   max_rows <- as.numeric(max_rows)
   verifyInput(max_rows, sorting)
-  if(exists("preprocessed")){
-    df <- preprocessed
-  }
-  else{
-    df <- mergeFetchedData(loaded_variables)
+  df <- parseInput()
+  df["Row.Label"] <-
+    lapply(df["Row.Label"],fixString)  # remove illegal
+                                       # characters from probe names. This will
+                                       # prevent problems with CSS
+                                       # selectors on the frontend.
+  write.table(
+      df,
+      "heatmap_orig_values.tsv",
+      sep = "\t",
+      na = "",
+      row.names = FALSE,
+      col.names = TRUE
+    )
+  df <- addStats(df, sorting)
+  df <- df[1:min(max_rows,nrow(df)),]  #  apply max_rows
+  fields <- buildFields(df)
+  extraFields <- buildExtraFields(fields)
+  probes <- na.omit(df[,1])
+  geneSymbols <- df[,2][1:length(probes)]
```

Comment:
This does not get the result you go for. na.omit on df does not change df. 
* Take a data.frame df with 5 rows, for which on the second row the value of the Row.Label column is missing (see example below). 
some_df<-data.frame(Row.Label = c("a", NA, "b","c","d"), Bio.Marker = c("a2", NA,"b2", "c2","d2" ), stringsAsFactors = F)
> some_df
  Row.Label Bio.Marker
1         a         a2
2      <NA>       <NA>
3         b         b2
4         c         c2
5         d         d2
* Then if you remove missing values from the probes, the vector with probes is shorter than the number of rows in the data.frame. It will have length 4.
* df[,2] [ 1:length(probes)]  will only select the values for bio.marker for the first 4 rows: result geneSymbols will contain value NA, but not the value from the last row

> probes <- na.omit(some_df[,1])
> geneSymbols <- some_df[,2][1:length(probes)]
> geneSymbols
[1] "a2" NA   "b2" "c2"

Also: row selection can be done like this than df[1:length(probes),2]  without the extra [] in df[,2][1:length(probes)]